app/controllers/close_contacts_controller.rb

```diff
@@ -42,4 +42,11 @@ def update
                                created_by: current_user.email,
                                comment: "User edited a close contact (ID: #{cc.id}).")
   end
+
+  # Delete an existing close contact record
+  def destroy
+    redirect_to root_url unless current_user.can_create_patient_close_contacts?
+    cc = CloseContact.find_by(id: params.permit(:id)[:id])
+    cc.destroy
```

Comment:
This works, but I'm a little concerned about the scoping of what this is allowed to delete.

I would prefer to see something more along the lines of (untested, also requires one additional param):
```
current_user.viewable_patients.find_by(id: params.permit(:patient_id)[:patient_id])&.close_contacts.find_by(id: params.permit(:id)[:id])&.destroy
```

This way you're scoping down by patients the current user is allowed to see, and then scoping down by the close contacts included on those patients.

Reply:
that makes sense. I'm still working on updating it with the archiving functionality, but I've pushed up a (slightly modified for functionality) version of that change that should fix the scoping issue.